src/renderer/navigation.ts

```diff
@@ -18,6 +18,7 @@ if (ipcRenderer) {
 }
 
 export function navigate(location: LocationDescriptor) {
+  if (history.location.pathname === location) return;
```

Comment:
This won't work when `location` from arguments is not a `string` (which is only part of `LocationDescriptor`)
Also it's not considering search-params and other parts of url. 

```suggestion
  const currentLocation = navigation.getPath();
  navigation.push(location);
  if (currentLocation === navigation.getPath()) {
    navigation.goBack(); // prevent sequences of same url in history
  }
```